File: TP2/Python/Punto4/CESAR/Profesor.py
class Profesor:
    def __init__(self, nombre,apellido):
        self.__nombre = nombre
        self.__apellido = apellido
        self.__Materia = []
    
    @property
    def nombre(self):
        return self.__nombre
    
    @property
    def apellido(self):
        return self.__apellido
    
    @nombre.setter
    def set_nombre(self,nombre):
        self.__nombre = nombre
    
    @apellido.setter
    def set_apellido(self,apellido):
        self.__apellido = apellido
    
    def agregar_Materia(self,Materia):
        self.__Materia.append(Materia)
    
    def imprimir(self):
        print("Profesor: ",self.__apellido,self.__nombre)
        print("Materias:")
        for pepe in self.__Materia:
            pepe.Imprimir()
            
        
    

File: TP2/Java/Punto 4/punto4 mora/Profesor.java
import java.util.ArrayList;

public class Profesor {
    private String nombre="";
    private String apellido="";
    private ArrayList<Materia> materias = new ArrayList<>();

    public Profesor (String nombre, String apellido){
        this.nombre = nombre;
        this.apellido = apellido;
    }

    public void setNombre(String nombre){
        this.nombre = nombre;
    }

    public String getNombre(){
        return nombre;
    }

    public void setApellido(String apellido){
        this.apellido = apellido;
    }

    public String getApellido(){
        return apellido;
    }

    public void setMateria (ArrayList <Materia> materia){
        this.materias = materia;
    }

    public ArrayList<Materia> getMateria(){
        return materias;
    }

    public void addMateria (Materia materia){
        this.materias.add(materia);
    }
}
